registry/txt.go

```diff
@@ -215,7 +219,7 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 
 	endpoints := make([]*endpoint.Endpoint, 0)
 
-	if !im.mapper.recordTypeInAffix() && r.RecordType != endpoint.RecordTypeAAAA {
+	if !im.mapper.recordTypeInAffix() && im.txtFormat == "transition" && r.RecordType != endpoint.RecordTypeAAAA && r.RecordType != endpoint.RecordTypeCNAME {
```

Comment:
I'm not sure we should suppress these for type CNAME. I've seen cases where the "new" format wasn't created because the domain got too long. The old format was created because the CNAME was really a Route53 alias record. Perhaps we just need to make this dependent on #3910.